internal/service/worker.go

```diff
@@ -135,32 +135,30 @@ func (w *Worker) Process(
 		storageBytes := storage.Bytes()
 		result = io.NopCloser(storage)
 
-		if w.BypassStoragePut.Bypass(ctx) {
-			return nil
-		}
-
-		baseSpan, ok := ddTracer.SpanFromContext(ctx)
-		if !ok {
-			return fmt.Errorf("fail to get span from context: %w", err)
-		}
-
-		storageSpan, storageCtx := ddTracer.StartSpanFromContext(
-			context.Background(), "Worker.Storage.Put", ddTracer.ChildOf(baseSpan.Context()),
-		)
-		go func() {
-			var err error
-			defer func() { storageSpan.Finish(ddTracer.WithError(err)) }()
-
-			if err = w.Storage.Put(storageCtx, hash, bytes.NewBuffer(storageBytes)); err != nil {
-				logger, nerr := w.TraceExtractor(storageCtx, w.Logger)
-				if nerr != nil {
-					logger.Err(nerr).Msg("Fail to extract the trace ID from the context")
-				}
-
-				err = fmt.Errorf("fail to put the object into the storage: %w", err)
-				logger.Err(err).Msg("Storage put error")
+		if !w.BypassStoragePut.Bypass(ctx) {
```

Comment:
Nice catch!

Instead of nesting the whole block inside the if we could simply do this, right?
```
if w.BypassStoragePut.Bypass(ctx) {
  return result
}
```

Reply:
We can't do this I'm afraid, we'd need to run an `io.Copy` as the output `io.Writer` is passed to the function. The function just returns an error.

Reply:
Ok, got it.  I was not with the whole function open. I'll approve this PR but I'll ask you just one thing, can you deploy it to dev and test everything in dev? dropbox and the e2e frontend tests, if everything is green we can merge to master and deploy to prod, ok?